### ISSUE TITLE
MacOS support

### ISSUE THREAD

**Issue by ianlewis** [google/gvisor#1270]
gVisor running on MacOS could provide an environment for Mac users to develop and run Linux containers on a Mac machine without needing to install Linux via a VM or Docker for Mac etc.

Some investigation needs to happen in order scope out what would be needed.

**Comment by psycadet** [comment]
@ianlewis  are you sure this would work? Linux syscalls could be translated into native Darwin syscalls by gvisor, but I think the host kernel (in this case Darwin) needs to have cgroups in order for this to work in the first place.

**Comment by ianlewis** [comment]
@psycadet Yeah, cgroups specifically may not be able to be supported natively. It may be that a virtualized shim (e.g. HyperKit) is needed like Docker for Mac or there may be other ways to make it work natively on a Mac. There will be a lot of challenges to be sure, but it would be an interesting feature if someone wanted to try and support it.

**Comment by ianlewis** [comment]
@mauriciogg  @kevinGC has been looking at it and has done the work to get us closer to full Mac support so he's the one to ask.

**Comment by l1n** [comment]
Interested in this again, if it's still being worked on!

**Comment by kevinGC** [comment]
It's not currently, unfortunately. No reason it can't be done, it's just that nobody has had the time to do it.

**Comment by maci0** [comment]
https://github.com/maci0/gvisor/blob/master/README-MACOS.md work in progress repo here.

**Comment by maci0** [comment]
It's mostly working now. there might be edge cases and cli parameters etc i'm missing but the core works 

```bash
➜  _tmp git:(master) ✗ uname -a
Darwin mwysocki-mac 25.4.0 Darwin Kernel Version 25.4.0: Thu Mar 19 19:33:25 PDT 2026; root:xnu-12377.101.15~1/RELEASE_ARM64_T6041 arm64
➜  _tmp git:(master) ✗ ./sentrydarwin --net=vmnet --rootfs alpine-rootfs /bin/sh
/bin/sh: can't access tty; job control turned off
/ # uname -a
Linux gvisor-darwin 4.4.0 #1 SMP Sun Jan 10 15:06:54 PST 2016 aarch64 Linux
/ # cat /etc/os-release
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.21.3
PRETTY_NAME="Alpine Linux v3.21"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://gitlab.alpinelinux.org/alpine/aports/-/issues"
/ #
```

**Comment by huyz** [comment]
@maci0 macOS 26 required, right? Not compatible with macOS 15?

**Comment by maci0** [comment]
@huyz I never tested with macOS 15.
